### PR TITLE
feat(tooling): lint code before pushing

### PR DESCRIPTION
**Pull request by alveshelio**
The goal of this PR is to add proper linting whenever a PR is pushed to make sure we have some standards.

Since the example relies on the lib folder that needs to be built every time we want to test our changes, I've added a script that will build the package whenever there's a change in any of the files within `src` directory.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        browser: true,
+        node: true,
+        es6: true,
+      },
+      extends: [
+        'eslint:recommended',
+        'plugin:@typescript-eslint/recommended',
+        'plugin:react-hooks/recommended',
+        'prettier/@typescript-eslint',
+        'plugin:prettier/recommended',
```
Comment (eemeli) [inline .eslintrc.js:26]:
```suggestion
        'prettier',
```

This simpler config is the current suggested way to combine ESLint + TS + Prettier.

**Review by eemeli — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (alveshelio) [reply .eslintrc.js:26]: From the ressources I've consulted to have prettier with eslint they suggest to do it with `plugin:prettier/recommended`
Here are some of the sources I've checked:
https://dev.to/caelinsutch/setting-up-a-typescript-nodejs-application-with-prettier-and-eslint-53jc
https://www.robertcooper.me/using-eslint-and-prettier-in-a-typescript-project
https://paulintrognon.fr/blog/typescript-prettier-eslint-next-js

Comment (eemeli) [reply .eslintrc.js:26]: Those might be outdated. Here are the `typescript-eslint` instructions for this: https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/README.md#usage-with-prettier

Comment (alveshelio) [reply .eslintrc.js:26]: Sweet, thank you for providing an update ressource :)

Comment (alveshelio) [reply .eslintrc.js]: If you are good with the PR you can go ahead and merge it. Let me know if there's anything else

**Review by eemeli — APPROVED** [review]

**Pull request merged**